rxbindings/src/main/java/com/amplifyframework/rx/RxDataStore.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.rx;
+
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.query.predicate.QueryPredicate;
+import com.amplifyframework.datastore.DataStoreCategoryBehavior;
+import com.amplifyframework.datastore.DataStoreItemChange;
+
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+
+/**
+ * An Rx-idiomatic expression of the behaviors in {@link DataStoreCategoryBehavior}.
+ */
+@SuppressWarnings("unused") // These are all public APIs
+public interface RxDataStore {
```

Comment:
`RxDataStoreCategoryBehavior`

Reply:
Eh, yea, I think you're probably right.

We had talked at one point about dropping the `CategoryBehavior` from even core. They're descriptive, but when you think about it from a customer standpoint, they have like:
```
DataStoreCategoryBehavior ds = Amplify.DataStore;
```
But the customer would probably rather just write like:
```
DataStore ds = Amplify.DataStore
```

Reply:
I went with the + `CategoryBehavior` versions for now. We can come back and remove them from all places at the same time, maybe.